### DISCUSSION TITLE
Convert Face Descriptor to Face Embedding

### DISCUSSION THREAD

**Discussion by Le0X8** [vladmandic/human#348] [Q&A] [answered]
Hi :)

Is there a way to convert Face Descriptors from face-api.js to Human's Embeddings?

What I mean with Face Descriptor: (await faceapi.detectAllFaces(tfImage(jpegBuffer).withFaceDescriptors())[0].descriptor (128 entries long Float32Array)

What I mean with Face Embedding: (await human.detect(tensor)).face[0].embedding (1024 entries long Array (idk if it's a Float32Array or not))

Why?
  -> I want to compare faces detected by face-api.js with these detected by Human

I have no experience and knowledge with AI or Tensors :/
I also asked ChatGPT about that, but it only responded some code snippets to me that cause lots of errors :(

Would be nice if someone could help me with that one, too :)

**Comment by vladmandic** [comment, marked as answer]
although they do the same thing, their content is very different and not comparable.
tensor compatibility could be solved (normalization, dimensionality, etc.), but the fact is that model output is just not comparable.
